src/views/CredentialsMenu/index.jsx

```diff
@@ -56,15 +57,7 @@ class CredentialsMenu extends React.PureComponent {
 
     return (
       userSession ? (
-        <Button
-          size="small"
-          variant="contained"
-          color="secondary"
-          className={classes.button}
-          onClick={() => context.setUserSession(null)}
-        >
-          Sign out
-        </Button>
+        <ProfileMenu userSession={userSession} context={context} />
```

Comment:
Don't pass down `userSession` but internally use `const userSession = context.getUserSession();`

You also don't need to pass the context as a prop. You can use this inside of `ProfileMenu`:
```javascript
import AuthContext from '../../components/auth/AuthContext';
...
class ProfileMenu extends React.Component {
   static contextType = AuthContext;`
```

You can then access it with `this.context` instead of a prop. You can read about it in here: https://reactjs.org/docs/context.html

Reply:
Thank you, it is interesting :)